ui/src/scenes/Schema/Table/ContextualMenu/services.ts

```diff
@@ -0,0 +1,38 @@
+import * as QuestDB from "utils/questdb"
+import { trim } from "ramda"
+
+export const formatTableSchemaQueryResult = (
+  name: string,
+  result: QuestDB.QueryRawResult,
+): string => {
+  if (result.type === QuestDB.Type.DQL) {
+    let query = `CREATE TABLE ${name} (`
+
+    for (let i = 0; i < result.count; i++) {
+      const column = result.dataset[i]
+      const name = column[0]
+      const typeDef = column[1]
+      const indexed = column[2]
+      const indexBlockCapacity = column[3]
+      const symbolCached = column[4]
+      const symbolCapacity = column[5]
+
+      query += `${name} ${typeDef} `
+      query += symbolCapacity ? `capacity ${symbolCapacity} ` : ""
+      query += symbolCached ? "cache " : "nocache "
```

Comment:
This adds `nocache` even if caching on this column is not supported. As far as I know, only the `symbol` column can have any of these additional params.

Reply:
> The designated timestamp column is not copied over when replicating the schema, i.e. manually creating a table with the following SQL:
> 
> ```sql
> CREATE TABLE my_schema_table (ts TIMESTAMP, int_col int) timestamp(ts)
> ```
> 
> When copying via this feature, it yields:
> 
> ```sql
> CREATE TABLE my_schema_table (ts TIMESTAMP nocache, int_col INT nocache)
> ```
> 
> Deviation being:
> 
> * `nocache` where not necessary
> * designated timestamp declaration

Hey @bsmth , the query I am using to fetch some details about the table does not return any information about timestamp. Do all tables have a timestamp define by default ? 

The query I am using is ```.queryRaw(`table_columns('${name}')`)```, please advise me on how to gather the timestamp, otherwise I can't really add it to the generated CREATE query

Reply:
committed a change to resolve that, only typeDef SYMBOL will now append those parameters

Reply:
tested in a36760db9cc0a05a1616f30a10c534ee07338e83 and looks good 👍🏻